### ISSUE TITLE
Support for composition api?

### ISSUE THREAD

**Issue by jaredmcateer** [ngVue/ngVue#102]
In preparing for Vue 3 I am trying to utilise the [@vue/composition-api](https://github.com/vuejs/composition-api) package but it seems ngVue doesn't recognize this api. Here's my component

```ts
<template>
  <div class="ps">
    <div class="ps__spinner" :class="{ '.ps__spinner--spin': inProgress }"></div>
  </div>
</template>

<script lang="ts">
import { createComponent, computed } from "@vue/composition-api";
import store from "@client/store";

export default createComponent({
  setup() {
    const inProgress = computed(() => store.getters.standBy.inProgress);
    return { inProgress };
  },
});
</script>
```

Using 

```ts
import ProgressSpinner from './progress-spinner.vue';

angular
.directive('progressSpinner'. ['createVueComponent', c => c(ProgressSpinner)];
```

resulted in an error

```
logger.js:109 2020-02-05T21:03:23.027Z ERROR browser-console Error: [$injector:unpr] Unknown provider: [object Object]Provider <- [object Object]
https://errors.angularjs.org/1.7.5/$injector/unpr?p0=%5Bobject%20Object%5DProvider%20%3C-%20%5Bobject%20Object%5D
    at https://dev.arterys.com/vendors.js:15125:12
    at https://dev.arterys.com/vendors.js:19892:19
    at Object.getService [as get] (https://dev.arterys.com/vendors.js:20052:32)
    at https://dev.arterys.com/vendors.js:19897:45
    at Object.getService [as get] (https://dev.arterys.com/vendors.js:20052:32)
    at https://dev.arterys.com/vendors.js:143690:73
    at ngVueLinker (https://dev.arterys.com/vendors.js:143691:10)
    at Object.link (https://dev.arterys.com/vendors.js:143788:21)
    at https://dev.arterys.com/vendors.js:16352:18
    at invokeLinkFn (https://dev.arterys.com/vendors.js:26222:9) <progress-spinner>
```

I then tried 

```ts
import ProgressSpinner from './progress-spinner.vue';

angular
  .value('ProgressSpinner', ProgressSpinner)
  .directive('progressSpinner'. ['createVueComponent', c => c('ProgressSpinner')];
```

And that at least got angular happy but vue complained about this

```
logger.js:109 2020-02-05T20:54:05.380Z ERROR browser-console [Vue warn]: Property
 or method "inProgress" is not defined on the instance but referenced during 
render. Make sure that this property is reactive, either in the data option, or 
for class-based components, by initializing the property. See: 
https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

found in

---> <ProgressSpinner> at client/ui/progress-spinner.vue
       <Root>
```

So it seems that the setup function isn't being run or something. I would be happy to work on this with anyone willing to guide me since it seems that you've moved on from this project. I'm just hoping to get a little more out of it since the migration of angular to vue is probably going to take us right up to end of life of angular. Thanks!

**Comment by khmyznikov** [comment]
TS:
```javascript
import { defineComponent } from '@vue/composition-api'

const _component = defineComponent({ /.../ });
const ProgressSpinnerComponent = Vue.component('progress-spinner', _component);
export default ProgressSpinnerComponent ;

angular
  .module('vue.components.progress-spinner', ['ngVue'])
  .directive('progressSpinner', function(createVueComponent) {
      return createVueComponent(ProgressSpinnerComponent);
  });
```

**Comment by jaredmcateer** [comment]
Fixed in #140
